src/any_agent/frameworks/any_agent.py

```diff
@@ -235,23 +239,33 @@ def serve(self, serving_config: A2AServingConfig | None = None) -> None:
             ```
 
         """
-        from any_agent.serving import A2AServingConfig, _get_a2a_app, serve_a2a
+        from any_agent.serving import serve_a2a
 
-        if serving_config is None:
-            serving_config = A2AServingConfig()
+        serve_a2a(self, serving_config)
```

Comment:
Would it be possible to remove the changes you made around a2a and consolidate this PR to only focus on serving over MCP? I'm happy to accept a re-org of the a2a changes but this PR is about MCP, and I'd rather take care of a2a related changes somewhere else.

Reply:
This arch aspect (single serving infra vs separate methods) was one of the items I wanted to agree on.